app/Hutch.Relay/Services/ResultsService.cs

```diff
@@ -46,53 +52,102 @@ public async Task SubmitResults(RelayTaskModel relayTask, JobResult jobResult
-        aggregateCount += result.Results.Count;
-      }
+      var finalResult = await PrepareFinalJobResult(task);
+
+      // Only submit results if Prepare succeeded
+      await SubmitResults(task, finalResult);
+    }
+    catch (ArgumentOutOfRangeException e)
+    {
+      // Catch and log, but otherwise no particular handling for Unsupported Task Types
+      logger.LogError(e.Message);
+    }
+    finally
+    {
+      // We should try and close out the task regardless of whether an exception occurred
+      await relayTaskService.SetComplete(task.Id);
     }
+  }
 
-    return new JobResult()
+  /// <summary>
+  /// Aggregates Sub Task results and obfuscates aggregate values. The details of this behaviour are specific to a given Task Type,
+  /// by means of an <see cref="IQueryResultAggregator"/> implementation.
+  ///
+  /// Adds the final aggregate results to a <see cref="JobResult"/> payload for the RelayTask, suitable to submit upstream.
+  /// </summary>
+  /// <param name="relayTask"><see cref="RelayTaskModel"/> providing details of the RelayTask to prepare a <see cref="JobResult"/> for.</param>
+  /// <returns>The <see cref="JobResult"/> containing aggregated and obfuscated data from all Sub Tasks.</returns>
+  /// <exception cref="ArgumentOutOfRangeException">The Task Type of this <see cref="RelayTaskModel"/> isn't supported by Relay. Who knows how it got this far.</exception>
+  public async Task<JobResult> PrepareFinalJobResult(RelayTaskModel relayTask)
+  {
+    // Get all SubTasks for this RelayTask
+    var subTasks = (await relayTaskService.ListSubTasks(relayTask.Id, incompleteOnly: false)).ToList();
+
+    // Select the appropriate Results Aggregator
+    IQueryResultAggregator aggregator = relayTask.Type switch
     {
-      Uuid = relayTaskId,
-      CollectionId = options.CollectionId ??
-                     throw new ArgumentException(nameof(options.CollectionId)),
-      Results = new QueryResult()
-      {
-        Count = aggregateCount,
-      }
+      TaskTypes.TaskApi_Availability => availabilityAggregator,
+      _ => throw new ArgumentOutOfRangeException(
+        $"Relay tried to handle a Task Type it doesn't support Results Aggregation for: {relayTask.Type}")
+    };
+
+    // Set the correct upstream values in the JobResult, and set the aggregate, obfuscated QueryResult data.
+    return new()
+    {
+      Uuid = relayTask.Id,
+      CollectionId = relayTask.Collection,
+      Results = aggregator.Process(subTasks)
     };
   }
 
+  /// <summary>
+  /// Check for incomplete RelayTasks past the timeout threshold and attempt to Complete them
+  /// </summary>
   public async Task HandleResultsToExpire()
   {
     var incompleteTasks = await relayTaskService.ListIncomplete();
     foreach (var task in incompleteTasks)
     {
-      logger.LogInformation("Task:{Task} is about to expire.", task.Id);
+      var expiryThreshold = TimeSpan.FromMinutes(1); // TODO: should this be configurable?
```

Comment:
Yes - but going from 4.5 minutes to 1 minute as default is also a big jump.
Distribution will take longer than 1 minute on any decent (100k+?) size dataset.
Feels like this is the kind of thing you might have changed for testing purposes

Not for now, but it'll also take more than 4.5 on anything much bigger I'd expect, so we should plan for this. Unless you have but I missed it in the 40 files!

Reply:
This was for testing, but:

- I think 4 for Availability is good
- I had not thought of distribution but let's implement 2 hours (like BC's defaults) in the next PR ;) for #47

Reply:
reverted / updated to 4 minutes